tests/integration-tests/tests/createami/test_createami.py

```diff
@@ -11,133 +11,211 @@
+    # Custom script
+    custom_script_file = "custom_script_ubuntu.sh" if os in ["ubuntu1804", "ubuntu2004"] else "custom_script.sh"
+
+    # Create S3 bucket for pre install scripts, to remove epel package if it is installed
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / custom_script_file), "scripts/custom_script.sh")
 
     # Get ParallelCluster AMI as base AMI
     base_ami = retrieve_latest_ami(region, os, ami_type="pcluster", architecture=architecture)
 
-    # Networking
-    vpc_id = vpc_stack.cfn_outputs["VpcId"]
-    networking_args = ["--vpc-id", vpc_id, "--subnet-id", vpc_stack.cfn_outputs["PublicSubnetId"]]
-
-    # Custom Cookbook
-    custom_cookbook = request.config.getoption("createami_custom_chef_cookbook")
-    custom_cookbook_args = [] if not custom_cookbook else ["-cc", custom_cookbook]
-
-    # Instance type
-    instance_args = ["-i", instance]
-
-    # Post install script
-    post_install_script_file = "post_install_ubuntu.sh" if os in ["ubuntu1804", "ubuntu2004"] else "post_install.sh"
-    post_install_script = "file://{0}".format(test_datadir / post_install_script_file)
-    post_install_args = ["--post-install", post_install_script]
-
-    pcluster_createami_result = run_command(
-        ["pcluster", "createami", "-ai", base_ami, "-os", os, "-r", region, "-c", cluster_config.as_posix()]
-        + custom_cookbook_args
-        + instance_args
-        + networking_args
-        + post_install_args,
-        timeout=7200,
+    image_id = "-".join(["integ", "test", "build", "image", "custom", "components", generate_random_string()])
+    cluster_config = pcluster_config_reader(
+        config_file="image.config.yaml",
+        parent_image=base_ami,
+        instance_type=instance,
+        bucket_name=bucket_name,
+        region=region,
     )
 
-    stdout_lower = pcluster_createami_result.stdout.lower()
-    assert_that(stdout_lower).does_not_contain("no post install script")
-    assert_that(stdout_lower).does_not_contain("no custom ami created")
+    result = build_image(image_id, region, cluster_config)
+
+    _assert_build_image_success(result, image_id, region)
+
 
+def _assert_build_image_success(result, image_id, region):
+    logging.info(f"Test build image process for image {image_id}.")
+    assert_that(result).contains("CREATE_IN_PROGRESS")
+    assert_that(result).contains("Build image started successfully.")
 
-@pytest.mark.dimensions("eu-central-1", "c5.xlarge", "ubuntu1804", "*")
-def test_createami_wrong_os(region, instance, os, request, pcluster_config_reader, vpc_stack, architecture):
-    """Test error message when os provide is different from the os of custom AMI"""
-    cluster_config = pcluster_config_reader()
+    pcluster_describe_image_result = run_command(["pcluster", "describe-image", "--id", image_id, "-r", region])
```

Comment:
here you should call image_factory, we should have something like:
```
image = image_factory(image_config) # in the main test function

image.describe() # here
```

By using this approach we can avoid the duplication of this describe-image command with all the parameter in multiple places and just let the factory manage them.